src/core/etl/src/Flow/ETL/Rows.php

```diff
@@ -177,6 +177,31 @@ public function entries() : array
         return $entries;
     }
 
+    public function equals(self $rows, Comparator $comparator = new NativeComparator()) : bool
+    {
+        if ($this->count() !== $rows->count()) {
+            return false;
+        }
+
+        foreach ($this->rows as $row) {
+            $found = false;
+
+            foreach ($rows->rows as $otherRow) {
+                if ($comparator->equals($row, $otherRow)) {
+                    $found = true;
+
+                    break;
+                }
+            }
+
+            if (!$found) {
+                return false;
+            }
```

Comment:
Probably can be simplified to:
```suggestion
            foreach ($rows->rows as $otherRow) {
                if ($comparator->equals($row, $otherRow)) {
                    return true;
                }
            }

            return false;
```

Reply:
I think I'm going to revert that change. I did it when I got stuck with comparing XML's, but it turned out that the process of canonicalization of XMLs does not remove whitespaces and line breakers between nodes. 
That method could still be very useful but now when I think about it deserves a separate pull request. However, your suggestion looks valid (but only tests can confirm that), will try to remember about it 😁